src/build.rs

```diff
@@ -49,11 +49,40 @@ fn rustc_minor_version() -> Option<u32> {
     otry!(pieces.next()).parse().ok()
 }
 
+fn check_wasm32_target() -> Result<bool, Error> {
+    let echo_cmd = Command::new("echo")
+        .arg("fn main(){}")
+        .stdout(Stdio::piped())
+        .spawn()
+        .context("failed to spawn `echo` when checking wasm32-unknown-unknown target")?;
```

Comment:
Why do you need to spawn a command here? Shouldn't you be able to just open a pipe and write into it?

Reply:
Yeah sorry @drager, I wasn't very clear in the comment on the original PR - by "equivalent of..." I didn't mean to literally invoke `echo`, but rather to use piping to `rustc` to write string directly.

You can do that with `.stdin(Stdio::piped())` and then writing a string.

Reply:
It's about as verbose either way, but yeah this is probably more portable.

Reply:
Oh, sure, it's likely even more verbose, but definitely more portable.

Reply:
Btw, one more thing: by default this will create a file called `rust_out` in the current dir. We might want to set `-o` to a file under OUT_DIR or something.

Reply:
My bad, I totally missed that a file was created... Thanks for letting me know. 

Also, I appreciate all the other comments from you all! What's your preference? Piping the string directly or invoke `echo`?

Reply:
Piping is more reliable than `echo` command, so I'd change it.